pppd/ipcp.c

```diff
@@ -1716,7 +1726,12 @@ ip_demand_conf(int u)
     if (!sifnpmode(u, PPP_IP, NPMODE_QUEUE))
 	return 0;
     if (wo->default_route)
+#ifndef __linux__
```

Comment:
This is ugly, having an #ifndef inside the function. Why not just add the extra parameter to the code in sys-solaris.c as well as sys-linux.c, and make the solaris version return an error if it is set?

Reply:
Indeed, fixed so.